Sandbox

nd-r opened the pull request.

8coon commented on css/style.css:
```diff
@@ -0,0 +1,52 @@
+body {
+    background: black round;
```
Зачем вам тут round?

8coon commented on css/style.css:
```diff
@@ -0,0 +1,52 @@
+
+.chalk-border {
+    border: 10px solid rgba(255,255,255,.8);
+    border-image: url("https://www.unicefusa.org/sites/default/files/answer-box.png") 20;
```
Урлы на внешние ресурсы это плохо

8coon commented on index.html:
```diff
@@ -0,0 +1,15 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <link href="//fonts.googleapis.com/css?family=Cabin+Sketch" rel="stylesheet" type="text/css">
```
То же самое, лучше нужный шрифт выкачать и отдавать из своих ресурсов

8coon commented on css/style.css:
```diff
@@ -0,0 +1,52 @@
+    width: 95%;
+    height: 15%;
+    left: 1%;
+}
```
Обычно принято оставлять последнюю строку в файле пустой

8coon commented on index.html:
```diff
@@ -0,0 +1,15 @@
+    <div class="footer-menu chalk-border"></div>
+</div>
+</body>
+</html>
```
Тут тоже нужна пустая строка

8coon commented on nodeServer/index.js:
```diff
@@ -1,17 +1,13 @@
-app.listen(process.env.PORT || 8080, () => {
+app.listen(app.get('port'), () => {
     console.log('Started');
 });
```
И тут пустая строка тоже нужна

avtyul commented on .gitignore:
```diff
@@ -1,2 +1,4 @@
 node_modules
 .idea
+package-lock.json
```
точно надо его игнорить ? (тут спорный вопрос просто, есть два стула)

ArturBmstu replied:
Я как-то безосновательно закинул его в гитигнор

8coon commented on nodeServer/index.js:
```diff
@@ -1,17 +1,69 @@
+        console.log(session_id);
+        res.json({error: null});
+    }else {
+        const error = 'USERNAME_ALREADY_EXISTS';
```
Понятно, что это временный сервер, но тем не менее: у вас где-то названия ошибок капсом и через_вот_этот_символ, а где-то просто строки. Надо бы унифицировать и все в константы вынести.

8coon commented on nodeServer/index.js:
```diff
@@ -1,17 +1,69 @@
+    }else {
+        const error = 'USERNAME_ALREADY_EXISTS';
+        console.log(error);
+        res.status(409).json({error:error});
```
{error}

8coon commented on nodeServer/index.js:
```diff
@@ -1,17 +1,69 @@
+app.post('/auth/me', (req, res) => {
+    const id = res.cookies['id'];
+
+    if(id){
```
Пробельчики вокруг (id)

8coon commented on nodeServer/index.js:
```diff
@@ -1,17 +1,69 @@
+    if(id){
+        const username = signedUsers[id];
+        res.json(username);
+    }else{
```
Пробельчики вокруг else

8coon commented on static/css/form-login.css:
```diff
@@ -0,0 +1,151 @@
+    display: none;
+}
+
+form.login {
```
Как-то очень похожи эти стили на то, что в файле form-add-username.css по селектору form.add-username. Вынесите их в отдельный файл и отдельный класс, и выставляйте этот класс в числе прочих у элемента, которому эти стили нужны.

8coon commented on static/css/form-signup.css:
```diff
@@ -0,0 +1,82 @@
+    margin-bottom: 20px;
+}
+
+form.signup input[type=text], form.signup input[type=password], form.signup input[type=email] {
```
То же самое, эти стили отличаются от того, что в form-login по селектору form.login input[type=text], form.login input[type=password] всего на одно значение!

8coon commented on static/css/form-signup.css:
```diff
@@ -0,0 +1,82 @@
+    background-color: transparent;
+}
+
+form.signup input[type=submit] {
```
И стили для submit тоже нужно переиспользовать. Это вообще некруто плодить одинаковый код.

8coon commented on static/js/app/startGame.js:
```diff
@@ -0,0 +1,24 @@
+        event.preventDefault();
+        user.whoami();
+        if (user.isAuthorized()) {
+            playGamePage.hide();
```
То, что я уже говорил -- управление страницами прямо из обработчика клика это плохо

8coon commented on static/js/app/startGame.js:
```diff
@@ -0,0 +1,24 @@
+
+export const startGame = () => {
+    const playGamePage = new Page(document.getElementById("play-game"));
+    playGamePage.elem.addEventListener('click', (event) => {
```
Эта функция точно вызовется единожды? Если нет, то кто тогда уберёт listener с playGamePage.elem?

8coon commented on static/js/auth/signUpNewUser.js:
```diff
@@ -0,0 +1,137 @@
+    LOGIN_PATH: '/auth/login'
+};
+
+const signUpForm = document.getElementById("signup").getElementsByTagName('form')[0];
```
`document.querySelector('#signup form')`

8coon commented on static/js/http/http.js:
```diff
@@ -0,0 +1,51 @@
+'use strict';
+
+export class http {
```
Так вы определитесь, вы все запросы через класс http отправляете, или через json_request?

8coon commented on static/js/tools/json_request.js:
```diff
@@ -0,0 +1,41 @@
+export const METHOD_POST = 'POST';
```
Эти константы вы где-то уже определяли. Да и не обязательно названия методов выносить в константы.

Commit:
fixed lint errors

Commit:
fixed lint errors[2]

zwirec committed:
Fix css

zwirec committed:
Merge remote-tracking branch 'origin/sandbox' into sandbox

zwirec committed:
Create Login Block Form

zwirec committed:
Create Login Block form

zwirec committed:
fix path

zwirec committed:
fix all

Commit:
fix

Commit:
fix

Commit:
deployment try

Commit:
mvc

Commit:
fix path

Commit:
added api

Commit:
rk2

Commit:
rk2

Commit:
rk2fix

Commit:
fix path

zwirec committed:
Update game.ts

zwirec committed:
Create .gitkeep

zwirec committed:
fix all

Commit:
Revert "fix all"

This reverts commit 2aa6b453ecfe82af64b229ec0e95335ab826b8b4.

Commit:
rk2 fix

Commit:
rk2 fix

Commit:
rk2 fix

Commit:
rk2 fix

Commit:
rk2 fix

Commit:
rk2 fix

Commit:
rk2 fix

Commit:
rk2 fix

Commit:
rk2 fix

zwirec committed:
fix

Commit:
rk2 fix

Commit:
rk2 fix

zwirec committed:
fix all

Commit:
pug?

Commit:
some changes

zwirec committed:
Merge remote-tracking branch 'origin/sandbox' into sandbox

# Conflicts:
#	static/js/game/gameLogic/gameState.ts

Commit:
added box2D dependency

Commit:
webpack fixes

Commit:
webpack fixes

Commit:
rm public.zip

Commit:
modified:   .gitignore
	deleted:    nodeServer/index.js
	deleted:    static/css/form-add-username.css
	deleted:    static/css/game-container.css
	deleted:    static/css/main.css
	deleted:    static/fonts/CabinSketch.woff2
	deleted:    static/fonts/CabinSketchBold.woff2
	deleted:    static/fonts/KGSecondChancesSketch.ttf
	deleted:    static/icons/facebook-logo.svg
	deleted:    static/icons/pointer-arrow-sketch-white.gif
	deleted:    static/icons/vkontakte-draw.svg
	deleted:    static/img/answer-box.png
	deleted:    static/img/border-chalk.png
	deleted:    static/img/chalk-background-filled.png
	deleted:    static/img/chalk_border_big.png
	deleted:    static/img/chalk_input_border.png
	deleted:    static/img/chalk_input_border_error.png
	deleted:    static/img/chalkboard.jpg
	deleted:    static/img/options_icon.svg
	deleted:    static/index.html
	deleted:    static/js/Offline/offline.ts
	deleted:    static/js/Offline/views/lobbyView/lobbyView.css
	deleted:    static/js/Offline/views/lobbyView/lobbyView.pug
	deleted:    static/js/Offline/views/lobbyView/lobbyView.ts
	deleted:    static/js/Online/online.ts
	deleted:    static/js/Online/views/lobbyView/lobbyView.css
	deleted:    static/js/Online/views/lobbyView/lobbyView.pug
	deleted:    static/js/Online/views/lobbyView/lobbyView.ts
	deleted:    static/js/Online/views/loginView/loginView.css
	deleted:    static/js/Online/views/loginView/loginView.pug
	deleted:    static/js/Online/views/loginView/loginView.ts
	deleted:    static/js/Online/views/signUpView/signUpView.css
	deleted:    static/js/Online/views/signUpView/signUpView.pug
	deleted:    static/js/Online/views/signUpView/signUpView.ts
	deleted:    static/js/application.ts
	deleted:    static/js/blocks/block.ts
	deleted:    static/js/blocks/button.js
	deleted:    static/js/blocks/form.js
	deleted:    static/js/blocks/input.js
	deleted:    static/js/blocks/loginForm.js
	deleted:    static/js/blocks/signUpForm.js
	deleted:    static/js/config/config.ts
	deleted:    static/js/config/paths.js
	deleted:    static/js/game/Box2D/Box2D.js
	deleted:    static/js/game/Box2D/Box2D.ts
	deleted:    static/js/game/Box2D/Collision/Shapes/b2ChainShape.js
	deleted:    static/js/game/Box2D/Collision/Shapes/b2ChainShape.ts

zwirec committed:
Merge remote-tracking branch 'origin/sandbox' into sandbox

# Conflicts:
#	static/js/game/body.ts
#	static/js/game/gameLogic/gameService.ts
#	static/js/views/gameView/gameView.ts

zwirec committed:
Fix game part

zwirec committed:
Fix game part

zwirec committed:
Fix game part

zwirec committed:
Fix game part

Commit:
fixes

Commit:
img fixes

Commit:
added logout

Commit:
fix

zwirec committed:
update path

Commit:
Merge branch 'sandbox' of https://github.com/frontend-park-mail-ru/2017_2_42 into sandbox

zwirec committed:
update devServer

Alexandera0 committed:
Merge branch 'sandbox' of https://github.com/frontend-park-mail-ru/2017_2_42 into sandbox

zwirec committed:
update wss

Commit:
Merge branch 'sandbox' of https://github.com/frontend-park-mail-ru/2017_2_42 into sandbox

zwirec committed:
update texture

Commit:
Merge branch 'sandbox' of https://github.com/frontend-park-mail-ru/2017_2_42 into sandbox

zwirec committed:
update wss

Commit:
Merge branch 'sandbox' of https://github.com/frontend-park-mail-ru/2017_2_42 into sandbox

Alexandera0 committed:
Merge branch 'sandbox' of https://github.com/frontend-park-mail-ru/2017_2_42 into sandbox

Commit:
Merge pull request #24 from Alexandera0/front

Front

Alexandera0 committed:
fixed

Commit:
Merge pull request #25 from Alexandera0/front

fixed

zwirec committed:
test service workers

Commit:
Merge branch 'sandbox' of https://github.com/frontend-park-mail-ru/2017_2_42 into sandbox

Commit:
Added some functionality

Commit:
added oauth

Commit:
Merge branch 'sandbox' of https://github.com/frontend-park-mail-ru/2017_2_42 into sandbox

Commit:
fixes scss

Commit:
Merge branch 'sandbox' of https://github.com/frontend-park-mail-ru/2017_2_42 into sandbox

Commit:
fixes scss

Commit:
Merge branch 'sandbox' of https://github.com/frontend-park-mail-ru/2017_2_42 into sandbox

Commit:
webpack fixes

Commit:
Merge branch 'sandbox' of https://github.com/frontend-park-mail-ru/2017_2_42 into sandbox

zwirec committed:
update service workers

zwirec committed:
Merge remote-tracking branch 'origin/sandbox' into sandbox

# Conflicts:
#	src/js/workers.js
#	webpack.config.js
#	webpack/devServer.js
#	webpack/lint.js

Commit:
Merge branch 'sandbox' of https://github.com/frontend-park-mail-ru/2017_2_42 into sandbox

Commit:
remove 'public'

zwirec committed:
Merge remote-tracking branch 'origin/sandbox' into sandbox

# Conflicts:
#	src/js/workers.js
#	webpack.config.js
#	webpack/devServer.js
#	webpack/lint.js

zwirec committed:
Merge remote-tracking branch 'origin/sandbox' into sandbox

Commit:
add .enc file

Commit:
update .travis.yml file

Commit:
update .travis.yml file

Commit:
update .travis.yml file

Commit:
update .travis.yml file

Commit:
update .travis.yml file

Commit:
update .travis.yml file

Commit:
update .travis.yml file

Commit:
update .travis.yml file

Commit:
fix login/signup server error response

Commit:
Merge remote-tracking branch 'origin/sandbox' into sandbox

Alexandera0 committed:
dobraya ya

Alexandera0 committed:
Merge branch 'sandbox' of https://github.com/frontend-park-mail-ru/2017_2_42 into sandbox

zwirec committed:
fix webpack and workers

zwirec committed:
Merge branch 'sandbox' of https://github.com/frontend-park-mail-ru/2017_2_42 into sandbox

Alexandera0 committed:
Merge branch 'sandbox' of https://github.com/frontend-park-mail-ru/2017_2_42 into sandbox

Commit:
update deploy

Commit:
Merge branch 'sandbox' of https://github.com/frontend-park-mail-ru/2017_2_42 into sandbox

Commit:
update deploy

Commit:
update deploy

Commit:
Added sound button

Commit:
Added lock on login/signup request

Commit:
refactor

Commit:
fixButton

Commit:
moar form fixes

Commit:
login form fix

Commit:
Canvas size at 4:3 scale

Commit:
update deploy

Commit:
Merge branch 'sandbox' of https://github.com/frontend-park-mail-ru/2017_2_42 into sandbox

Commit:
update

Alexandera0 committed:
update

Alexandera0 committed:
update

Commit:
Some optimizations

Commit:
Merge remote-tracking branch 'origin/sandbox' into sandbox

Alexandera0 committed:
update

Alexandera0 committed:
Merge branch 'sandbox' of https://github.com/frontend-park-mail-ru/2017_2_42 into sandbox

Commit:
Some fixes

Commit:
favicon added

Commit:
Merge pull request #27 from Alexandera0/sandbox

Dobraya

Alexandera0 committed:
Merge branch 'sandbox' of https://github.com/frontend-park-mail-ru/2017_2_42 into sandbox

Commit:
update

Commit:
Merge pull request #28 from Alexandera0/sandbox

Sandbox

Commit:
resized to 16x9

Commit:
Merge remote-tracking branch 'origin/sandbox' into sandbox

zwirec committed:
update

zwirec committed:
Merge branch 'sandbox' of https://github.com/frontend-park-mail-ru/2017_2_42 into sandbox

Commit:
deployment

Commit:
deployment[2]

Commit:
deployment[3]

Commit:
deployment[n]

Commit:
favicon added

Commit:
deployment[n+1]

Commit:
deployment[n+n]

Commit:
dontaskit

Commit:
favicon added

Commit:
fix

zwirec committed:
Merge branch 'sandbox' of https://github.com/frontend-park-mail-ru/2017_2_42 into sandbox

Alexandera0 committed:
update

Alexandera0 committed:
Merge branch 'sandbox' of https://github.com/frontend-park-mail-ru/2017_2_42 into sandbox

zwirec committed:
Merge pull request #30 from Alexandera0/sandbox

Sandbox

zwirec committed:
Merge branch 'sandbox' of https://github.com/frontend-park-mail-ru/2017_2_42 into sandbox

zwirec committed:
update

Alexandera0 committed:
update

Alexandera0 committed:
Merge branch 'sandbox' of https://github.com/frontend-park-mail-ru/2017_2_42 into sandbox

Commit:
typings fix

Alexandera0 committed:
«update»

Commit:
favicon added

Commit:
favicon added

Commit:
compressed img

Commit:
Update readme.md

Commit:
favicon added

Commit:
favicon added

Commit:
PushHistory fix

zwirec committed:
update

zwirec committed:
update

Commit:
Bug

Commit:
Bug

zwirec committed:
update

zwirec committed:
udpate

Commit:
Bug

zwirec committed:
update

zwirec committed:
Merge branch 'sandbox' of https://github.com/frontend-park-mail-ru/2017_2_42 into sandbox

Commit:
fix ball

Commit:
Bug

zwirec committed:
update

zwirec committed:
Merge branch 'sandbox' of https://github.com/frontend-park-mail-ru/2017_2_42 into sandbox

Alexandera0 committed:
update

zwirec committed:
update

Alexandera0 committed:
Merge branch 'sandbox' of https://github.com/frontend-park-mail-ru/2017_2_42 into sandbox

zwirec committed:
update

Alexandera0 committed:
update

zwirec committed:
Merge branch 'sandbox' into sandbox

zwirec committed:
Merge pull request #34 from Alexandera0/sandbox

Sandbox update

zwirec committed:
update

zwirec committed:
merge

zwirec committed:
fixed

Commit:
Username in lobby

Commit:
Sounds added

Commit:
Bugfix

Commit:
Bugfix[2]

Alexandera0 committed:
«update»

zwirec committed:
udpate

zwirec committed:
Merge branch 'sandbox' of https://github.com/frontend-park-mail-ru/2017_2_42 into sandbox

Alexandera0 committed:
«update»

Alexandera0 committed:
«update»

Commit:
music changed

Commit:
Merge remote-tracking branch 'origin/sandbox' into sandbox

zwirec committed:
fix lobby bug

zwirec committed:
Merge branch 'sandbox' of https://github.com/frontend-park-mail-ru/2017_2_42 into sandbox

Alexandera0 committed:
«update»

Alexandera0 committed:
«merge»

zwirec committed:
all bad

zwirec committed:
Merge branch 'sandbox' into sandbox

zwirec committed:
update

zwirec committed:
Merge pull request #35 from Alexandera0/sandbox

Sandbox

Commit:
workers redone

Commit:
Merge remote-tracking branch 'origin/sandbox' into sandbox

zwirec committed:
update

zwirec committed:
update

zwirec committed:
update

Commit:
Canvas responsive

Commit:
Merge remote-tracking branch 'origin/sandbox' into sandbox

# Conflicts:
#	src/js/game/gameLogic/gameOnline.ts
#	src/js/views/Pages/gameView/gameView.ts

zwirec committed:
ya sdelal

zwirec committed:
update

zwirec committed:
update

zwirec committed:
update

zwirec committed:
one more night...

zwirec committed:
since this commit send snaps

Alexandera0 committed:
«update»

zwirec committed:
increase size of rotating shtoochka

zwirec committed:
add payload to board message

zwirec committed:
fix fallthrough ball

Commit:
Merge branch 'sandbox' into sandbox

Commit:
Merge pull request #36 from Alexandera0/sandbox

«update»

Commit:
Fixed music + profile overlay

Commit:
Fixed removal of animation

Commit:
bug fix

Commit:
hotfix subscribe msg

Commit:
Overlay fixes

Commit:
returned back

Commit:
Profile finished

Commit:
Profile fix

zwirec committed:
friday commit

zwirec committed:
add some events

Commit:
Profile fix

Commit:
Deleted image from map tile

Commit:
img fix

zwirec committed:
update

zwirec committed:
Merge branch 'sandbox' of https://github.com/frontend-park-mail-ru/2017_2_42 into sandbox

Commit:
Almost final look

Commit:
Merge remote-tracking branch 'origin/sandbox' into sandbox

Commit:
Almost offline working

Commit:
View fixes

zwirec committed:
nine hours later...

zwirec committed:
Merge branch 'sandbox' of https://github.com/frontend-park-mail-ru/2017_2_42 into sandbox

zwirec committed:
vse

Commit:
Win lose fixes

Commit:
Final may be?

Commit:
Service workers up

Commit:
Service workers fix

Commit:
Service workers fix me path

zwirec committed:
uff

zwirec committed:
Merge branch 'sandbox' of https://github.com/frontend-park-mail-ru/2017_2_42 into sandbox

Commit:
fixes

Commit:
fixes

Commit:
fixes

Commit:
fixes
